lib/calculated_metrics.rb

```diff
@@ -47,34 +60,48 @@ def percent_rejected(prison, reason=nil)
 
   private
   def calculate_percentiles(column, percentile)
-    # This is a truly horrible hack. It lets us get past AR built-in value quoting, by pretending to be another object.
-    # It is needed so that we can parametrize two queries below by column, without resorting to putting the query together
-    # by hand. Don't call this method from outside this class, please.
+    # This is a truly horrible hack. It lets us get past AR built-in value
+    # quoting, by pretending to be another object. It is needed so that we
+    # can parametrize two queries below by column, without resorting to
+    # putting the query together by hand. Don't call this method from outside
+    # this class, please.
     column.define_singleton_method(:quoted_id) do
       column
     end
-    (@date_range ? calculate_percentiles_with_date_range(column, percentile) : calculate_percentiles_without_date_range(column, percentile)).each_with_object({}) do |row, h|
+    if @date_range
+      calculate_percentiles_with_date_range(column, percentile)
+    else
+      calculate_percentiles_without_date_range(column, percentile)
+    end.each_with_object({}) do |row, h|
       h[row['nomis_id']] = row[column].to_i
     end
   end
 
   def calculate_percentiles_with_date_range(column, percentile)
     @model.find_by_sql ['
-WITH percentiles AS (SELECT nomis_id, ?, cume_dist() OVER (PARTITION BY nomis_id ORDER BY ?) AS percentile
+WITH percentiles AS (SELECT nomis_id, ?, cume_dist()
+                      OVER (PARTITION BY nomis_id ORDER BY ?
+                    ) AS percentile
                      FROM visit_metrics_entries WHERE ? IS NOT NULL
                      AND requested_at > ?::date
                      AND processed_at <= ?::date),
-     top_percentiles AS (SELECT nomis_id, ?, rank() OVER (PARTITION BY nomis_id ORDER BY ?)
+     top_percentiles AS (SELECT nomis_id, ?, rank()
+                          OVER (PARTITION BY nomis_id ORDER BY ?)
                      FROM percentiles WHERE percentile >= ?)
 SELECT nomis_id, ? FROM top_percentiles WHERE rank = 1
```

Comment:
This SQL could be indented just like the rest of the code: it doesn't have to start at the first column.

Also, I think it's nice to use `<<-SQL` … `SQL` as a delimiter: some editors can then highlight the inner language properly:

```ruby
query = <<-SQL
  WITH percentiles AS (
    SELECT nomis_id, ?, cume_dist()
      OVER (PARTITION BY nomis_id ORDER BY ?
    ) AS percentile
  …
SQL

@model.find_by_sql [
  query, column, column, …
]
```